src/interfaces/toMarcRecords.js

```diff
@@ -47,6 +50,10 @@ export default function (amqpOperator) {
         .on('end', async () => {
           logger.log('info', `Read ${promises.length} records from stream`);
           logger.log('info', 'Sending records to queue! This might take some time!');
+          await setTimeoutPromise(500); // Makes sure that even slowest promise is in the array
+          if (promises.length === 0) { // eslint-disable-line functional/no-conditional-statement
+            reject(new ApiError(httpStatus.UNPROCESSABLE_ENTITY, 'Invalid payload!'));
+          }
```

Comment:
I'd return here, because reject doesn't necessarily terminate the function immediately.